### ISSUE TITLE
RedisCluster should support connecting to new nodes not in initial set of servers

### ISSUE THREAD

**Issue by klk-** [etaty/rediscala#177]
Currently the list of allowed server connections in RedisCluster is determined solely by the initial set of servers it is given.  However, it's possible during normal operations that a new node gets added to the cluster (e.g. due to regular maintenance).  In this case the RedisCluster needs to be recreated with an updated server list.

It would be much nicer if RedisCluster could gracefully respond to this type of situation by making new connections to nodes that it receives in ASK or MOVED responses.

**Comment by polymorpher** [comment]
Yes, even some periodical auto-discovery functionality would help a lot.

**Comment by polymorpher** [comment]
I have to switch back to Java client (lettuce) because of this. My project requires robust, dynamic cluster support. Perhaps we can implement something like what they do here https://github.com/lettuce-io/lettuce-core/wiki/Redis-Cluster
